lib/alipay_global/service/settlement.rb

```diff
@@ -0,0 +1,74 @@
+require 'open-uri'
+
+module AlipayGlobal
+  module Service
+    module Settlement
+
+      SETTLEMENT_REQUIRED_PARAMS = %w( start_date end_date )
+
+      SETTLEMENT_ERROR_MESSAGES = [
+        "Over 10 days to Date period",
+        "Over limit balance amount record",
+        "System exception",
+        "No balance amount data in the period",
+        "<alipay><is_success>F</is_success><error>ILLEGAL_PARTNER</error></alipay>",
+        "Finish date not ahead of today",
+        "Finish date ahead of begin date",
+        "Illegal Date period!",
+        "Date format incorrect YYYYMMDD",
+        "Internet connected exception ,please try later"
+      ]
+
+      def self.request(params)
+        settlement_resp = []
+        open(build_request(params)) do |data|
+          results = data.read
+
+          return false if results.include? "No balance account data in the period"
+          raise ArgumentError, "#{results}" if error_check(results)
```

Comment:
Is there a reason why we're raising `ArgumentError` here? `ArgumentError` suggest that wrong parameters were passed to function. I think we need some custom error here